specs/core/0_beacon-chain.md

```diff
@@ -1763,12 +1761,11 @@ def process_deposit(state: BeaconState, deposit: Deposit) -> None:
         leaf=hash_tree_root(deposit.data),
         proof=deposit.proof,
         depth=DEPOSIT_CONTRACT_TREE_DEPTH,
-        index=deposit.index,
+        index=state.deposit_index,
```

Comment:
we can now arbitrarily reorder unprocessed deposits as long as they are equivalent under `index // (2**i) % 2:` in the verification calculation... do we trust validators to present deposits in chronological order?